src/atomate2/abinit/sets/anaddb.py

```diff
@@ -0,0 +1,381 @@
+        """
+        return self.anaddb_input.remove_vars(keys=keys, strict=strict)
+
+    def set_structure(self, structure: Any) -> Structure:
+        """Set the structure for this input set.
+
+        This basically forwards the setting of the structure to the abipy
+        AnaddbInput object.
+        """
+        return self.anaddb_input.set_structure(structure)
+
+    def deepcopy(self) -> AnaddbInputSet:
+        """Deep copy of the input set."""
+        return copy.deepcopy(self)
+
+
+@dataclass
+class AnaddbInputGenerator(AbiBroadInputGenerator):
+    """
+    A class to generate Anaddb input sets.
+
+    Parameters
+    ----------
+    calc_type
+        A short description of the calculation type
+    user_anaddb_settings
+        A dictionary that allows to set any Abinit variable in the AbinitInput
+        after it has been generated from the factory function. This will override
+        any value or default previously set. Set a value to None to remove it
+        from the input.
+    user_kpoints_settings
+        Allow user to override kpoints setting by supplying a dict. E.g.,
+        ``{"reciprocal_density": 1000}``. User can also supply a KSampling object.
+    restart_from_deps:
+        Defines the files that needs to be linked from previous calculations in
+        case of restart. The format is a tuple where each element is a list of
+        "|" separated run levels (as defined in the AbinitInput object) followed
+        by a colon and a list of "|" list of extensions of files that needs to
+        be linked. The runlevel defines the type of calculations from which the
+        file can be linked. An example is (f"{NSCF}:WFK",).
+    prev_outputs_deps
+        Defines the files that needs to be linked from previous calculations and
+        are required for the execution of the current calculation.
+        The format is a tuple where each element is a list of  "|" separated
+        run levels (as defined in the AbinitInput object) followed by a colon and
+        a list of "|" list of extensions of files that needs to be linked.
+        The runlevel defines the type of calculations from which the file can
+        be linked. An example is (f"{NSCF}:WFK",).
+    factory_prev_inputs_kwargs
+        A dictionary defining the source of the of one or more previous
+        AbinitInput in case they are required by a factory to build a new
+        AbinitInput. The key should match the name of the argument of the factory
+        function and the value should be a tuple with the runlevels of the
+        compatible types of AbinitInput that can be used.
+    force_gamma
+        Force gamma centered kpoint generation.
+    symprec
+        Tolerance for symmetry finding, used for line mode band structure k-points.
+    """
+
+    factory: Callable | None = None
+    calc_type: str = "anaddb"
+    factory_kwargs: dict = field(default_factory=dict)
+    user_abinit_settings: dict = field(default_factory=dict)
```

Comment:
I let abinit because anaddb variables are considered as abinit variables (I think) and I think it is better to not confuse the user with different variables name for the same usage since it is all part of the ABINIT workflows